backend/inspirehep/migrator/tasks.py

```diff
@@ -109,6 +111,38 @@ def migrate_record_from_legacy(recid):
     db.session.commit()
 
 
+def migrate_from_mirror_run_step(
+    disable_orcid_push=True, disable_references_processing=True, step_no=1
+):
+    """Allows to easily run step by step migration only for valid records """
+    if step_no == 0:
+        query = LegacyRecordsMirror.query.with_entities(
+            LegacyRecordsMirror.recid
+        ).filter(LegacyRecordsMirror.valid.is_(True))
+    elif 0 < step_no < 3:
+        query = PersistentIdentifier.query.with_entities(
+            PersistentIdentifier.object_uuid
+        )
+    else:
+        echo("Wrong step number!")
+        return
+    if step_no == 0:
+        recids_chunked = chunker(str(res.recid) for res in query.yield_per(CHUNK_SIZE))
+    else:
+        recids_chunked = chunker(
+            str(res.object_uuid) for res in query.yield_per(CHUNK_SIZE)
+        )
```

Comment:
Just last thing, can we just reduce the ``if``s to the following?

```python

    if step_no == 0:
        query = LegacyRecordsMirror.query.with_entities(
            LegacyRecordsMirror.recid
        ).filter(LegacyRecordsMirror.valid.is_(True))
        recids_chunked = chunker(str(res.recid) for res in query.yield_per(CHUNK_SIZE))
    elif 0 < step_no < 3:
        query = PersistentIdentifier.query.with_entities(
            PersistentIdentifier.object_uuid
        )
        recids_chunked = chunker(
            str(res.object_uuid) for res in query.yield_per(CHUNK_SIZE)
        )
    else:
        echo("Wrong step number!")
        return
 
    task = migrate_recids_from_mirror(...)
```

Reply:
:+1: